### PR TITLE
Fix exporting info about failed e2e

### PR DESCRIPTION
**Pull request by sleshchenko**
### What issues does this PR fix or reference?

Previously, logs were saved after make uninstall is done. So nothing is saved
This PR fixes exporting info about failed e2e and logs are saved after script is interupted/exited or before `make uninstall`.
Also, now we export DevWorkspaces.

In addition, e2e tests better report failed DevWorkspaces. Now we the the following output:
```
------------------------------
• Failure [4.005 seconds]
[Create OpenShift Web Terminal Workspace]
/home/sleshche/projects/devworkspace-operator/test/e2e/pkg/tests/devworkspaces_tests.go:25
  Add OpenShift web terminal to cluster and wait running status [It]
  /home/sleshche/projects/devworkspace-operator/test/e2e/pkg/tests/devworkspaces_tests.go:42

  OpenShift Web terminal workspace didn't start properly. Error: workspace has been failed unexpectedly. Message: Error processing devfile: failed to read plugin for component web-terminal from internal registry: environment variable RELATED_IMAGE_plugin_redhat_developer_web_terminal_4_5_0 is unset              
```

### Is it tested? How?
I tested it locally and got 
```
/tmp/artifacts > tree .                                                                                              12:30:43
.
├── devworkspace-controller-info
│   ├── devworkspace-controller-manager-5fc9cbbc54-5xztv-devworkspace-controller.log
│   ├── devworkspace-controller-manager-6f6d74f498-6bxl6-devworkspace-controller.log
│   ├── devworkspace-webhook-server-774d8767cb-rxj76-webhook-server.log
│   └── events.log
├── devworkspaces.yaml
├── junit-workspaces-operator.xml
└── test-terminal-namespace-info
    └── events.log

```

It works on Prow and can be checked on https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/devfile_devworkspace-operator/383/pull-ci-devfile-devworkspace-operator-main-v7-devworkspaces-operator-e2e/1382234097992077312/artifacts/devworkspaces-operator-e2e/devworkspaces-operator-e2e/artifacts/

### CODE REVIEW

File: .ci/openshift_e2e.sh [hunk 1]
```diff
@@ -18,12 +18,37 @@ set -u
 # print each command before executing it
 set -x
 
-trap 'Catch_Finish $?' EXIT SIGINT
+function bumpPodsInfo() {
+    NS=$1
+    TARGET_FLD=${ARTIFACTS_DIR}/$NS-info
```
Comment (sleshchenko) [inline .ci/openshift_e2e.sh:23]: It's just a quick fix, if you have better folders/files structure proposal, let me know.

File: .ci/openshift_e2e.sh [hunk 2]
```diff
@@ -18,17 +18,45 @@ set -u
 # print each command before executing it
 set -x
 
-trap 'Catch_Finish $?' EXIT SIGINT
+function bumpPodsInfo() {
+    NS=$1
+    TARGET_FLD=${ARTIFACT_DIR}/$NS-info
```
Comment (amisevsk) [inline .ci/openshift_e2e.sh:23]:
```suggestion
    TARGET_FLD="${ARTIFACT_DIR}/${NS}-info"
```
Optional (for clarity)

Comment (amisevsk) [reply .ci/openshift_e2e.sh:23]: Also I still read `FLD` as `field` -- consider using `DIR`

**Review by amisevsk — APPROVED** [review]

**Comment by amisevsk** [comment]
Great improvement :)

**Review by JPinkney — APPROVED** [review]

File: .ci/openshift_e2e.sh (hunk 2 above)
Comment (sleshchenko) [reply .ci/openshift_e2e.sh]: Thanks. applied

**Comment by sleshchenko** [comment]
Tests are failed on deprovision steps, infra issues. But according to artifacts, everything was fine.

**Pull request merged**